linera-core/src/client/mod.rs

```diff
@@ -3236,16 +3233,11 @@ where
         &self,
         remote_node: RemoteNode<P::Node>,
     ) -> Result<(), ChainClientError> {
-        let mutex = self.state().client_mutex();
-        let _guard = mutex.lock_owned().await;
```

Comment:
TBH I don't know why we can remove it now. It protected from concurrent calls to `synchronize_received_certificates_from_validator` - why do we not care about that anymore?